src/db/models/merchandise.js

```diff
@@ -0,0 +1,183 @@
+const { pick } = require('underscore');
+
+module.exports = (Sequelize, db) => {
+  const Merchandise = db.define('Merchandise', {
+    id: {
+      type: Sequelize.INTEGER,
+      primaryKey: true,
+      autoIncrement: true,
+    },
+
+    uuid: {
+      type: Sequelize.UUID,
+      defaultValue: Sequelize.UUIDV4,
+    },
+
+    itemName: {
+      type: Sequelize.STRING,
+      allowNull: false,
+      validate: {
+        len: {
+          args: [2, 255],
+          msg: 'The item name must be at least 2 characters long',
+        },
+        notEmpty: {
+          msg: 'The item name is a required field',
+        },
+      },
+    },
+
+    collection: {
+      type: Sequelize.UUID,
+      allowNull: false,
+    },
+
+    picture: {
+      type: Sequelize.STRING,
+      validate: {
+        isUrl: true,
+      },
+    },
+
+    price: {
+      type: Sequelize.INTEGER,
+      allowNull: false,
+      validate: {
+        min: 0,
+      },
+    },
+
+    quantity: {
+      type: Sequelize.INTEGER,
+      defaultValue: 0,
+      validate: {
+        min: 0,
+      },
+    },
+
+    description: {
+      type: Sequelize.TEXT,
+      allowNull: false,
+    },
+
+    discountPercentage: {
+      type: Sequelize.INTEGER,
+      defaultValue: 0,
+      validate: {
+        min: 0,
+      },
+    },
+
+    monthlyLimit: {
+      type: Sequelize.INTEGER,
+    },
+
+    lifetimeLimit: {
+      type: Sequelize.INTEGER,
+    },
+
+    numSold: {
+      type: Sequelize.INTEGER,
+      defaultValue: 0,
+    },
+
+    hidden: {
+      type: Sequelize.BOOLEAN,
+      allowNull: false,
+      defaultValue: true,
+    },
+  }, {
+    timestamps: false,
+    indexes: [
+      // for lookup by UUID
+      {
+        using: 'BTREE',
+        unique: true,
+        fields: ['uuid'],
+      },
+
+      // for retrieving discounted inventory, sorted by highest discount
+      {
+        using: 'BTREE',
+        fields: [{ attribute: 'discountPercentage', order: 'DESC' }],
+        where: { discountPercentage: { [Sequelize.Op.gt]: 0 } },
+      },
+
+      // for retrieving all items in a collection
+      {
+        name: 'merchandise_collections_index',
+        using: 'BTREE',
+        fields: ['collection'],
+      },
+    ],
+    freezeTableName: true,
+    tableName: 'Merchandise',
+  });
+
+  Merchandise.findByUUID = function (uuid) {
+    return this.findOne({ where: { uuid } });
+  };
+
+  Merchandise.findAllByCollection = function (collection, isAdmin) {
+    const where = isAdmin ? { collection } : { collection, hidden: false };
+    return this.findAll({ where });
+  };
+
+  Merchandise.getDiscountedItems = function (offset, limit) {
+    if (!offset || offset < 0) offset = 0;
+    if (!limit || limit < 0) limit = undefined;
+    return this.findAll({
```

Comment:
Do you also need to check for `hidden: false` here for non-admin users?

Reply:
The check could be elsewhere, e.g. the API layer, but I don't think this function is being used so I'm just gonna delete it.